cardscan-base/src/main/java/com/getbouncer/cardscan/base/SSDOcrDetect.java

```diff
@@ -96,6 +100,63 @@ private String runModel(Bitmap image) {
         return number;
     }
 
+    public interface OnDetectListener {
+        void complete(String result);
+    }
+
+    private static <T> T getKeyByMaxValue(Map<T, ? extends Comparable> map) {
+        Map.Entry<T, ? extends Comparable> maxEntry = null;
+        for (Map.Entry<T, ? extends Comparable> entry : map.entrySet()) {
+            if (maxEntry == null || entry.getValue().compareTo(maxEntry.getValue()) > 0) {
+                maxEntry = entry;
+            }
+        }
+
+        if (maxEntry != null) {
+            return maxEntry.getKey();
+        } else {
+            return null;
+        }
+    }
+
+    public static void runInCompletionLoop(
+        final List<Bitmap> frames,
+        final Context context,
+        final OnDetectListener detection
+    ) {
+        new Thread(new Runnable() {
```

Comment:
so this starts a new background thread that runs the OCR model on the captured frames. And the background thread communicates to the main thread (`Looper.getMainLooper()`) after it finishes all of its predictions, via the `handler`?

Reply:
exactly!